packages/discord.js/src/managers/GuildSoundboardSoundManager.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const { lazy } = require('@discordjs/util');
+const { Routes } = require('discord-api-types/v10');
+const { CachedManager } = require('./CachedManager.js');
+const { DiscordjsTypeError, ErrorCodes } = require('../errors/index.js');
+const { SoundboardSound } = require('../structures/SoundboardSound.js');
+const { resolveBase64, resolveFile } = require('../util/DataResolver.js');
+
+const fileTypeMime = lazy(() => require('magic-bytes.js').filetypemime);
+
+/**
+ * Manages API methods for Soundboard Sounds and stores their cache.
+ * @extends {CachedManager}
+ */
+class GuildSoundboardSoundManager extends CachedManager {
+  constructor(guild, iterable) {
+    super(guild.client, SoundboardSound, iterable);
+
+    /**
+     * The guild this manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of Soundboard Sounds
+   * @type {Collection<Snowflake, SoundboardSound>}
+   * @name GuildSoundboardSoundManager#cache
+   */
+
+  _add(data, cache) {
+    return super._add(data, cache, { extras: [this.guild], id: data.sound_id });
+  }
+
+  /**
+   * Data that resolves to give a SoundboardSound object. This can be:
+   * * A SoundboardSound object
+   * * A Snowflake
+   * @typedef {SoundboardSound|Snowflake} SoundboardSoundResolvable
+   */
+
+  /**
+   * Resolves a SoundboardSoundResolvable to a SoundboardSound object.
+   * @method resolve
+   * @memberof GuildSoundboardSoundManager
+   * @instance
+   * @param {SoundboardSoundResolvable} soundboardSound The SoundboardSound resolvable to identify
+   * @returns {?SoundboardSound}
+   */
+
+  /**
+   * Resolves a {@link SoundboardSoundResolvable} to a {@link SoundboardSound} id.
+   * @param {SoundboardSoundResolvable} soundboardSound The soundboard sound resolvable to resolve
+   * @returns {?Snowflake}
+   */
+  resolveId(soundboardSound) {
+    if (soundboardSound instanceof this.holds) return soundboardSound.soundId;
+    if (typeof soundboardSound === 'string') return soundboardSound;
+    return null;
+  }
+
+  /**
+   * Options used to create a soundboard sound in a guild.
+   * @typedef {Object} GuildSoundboardSoundCreateOptions
+   * @property {BufferResolvable|Stream} file The file for the soundboard sound
+   * @property {string} name The name for the soundboard sound
+   * @property {string} [contentType] The content type for the soundboard sound file
+   * @property {number} [volume] The volume for the soundboard sound, from 0 to 1. Defaults to 1
```

Comment:
Is this inclusive of 0, so you can upload mute soundboard sounds? Might be wise to clarify this!